### PR TITLE
feat: wappalyzer: Handle simple DOM selectors

### PR DESCRIPTION
**Pull request by kingthorin**
Fixes zaproxy/zaproxy#6607

- Application.java > Add new field, setter, and getter.
- apps.json > Add new simple test entry.
- CHANGELOG.md > Added change note.
- WappalyzerJsonParser.java > Add handling for building/using a simple string list with DOM selector patterns.
- WappalyzerJsonParserUnitTest.java > Updated test assertions.
- WappalyzerPassiveScanner.java > Updated to handle the new simple DOM selector patterns/functionality.
- WappalyzerPassiveScannerUnitTest.java > Added a test method to test the new simple DOM selector functionality.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

File: addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java [hunk 1]
```diff
@@ -123,12 +122,13 @@ private void checkAppMatches(HttpMessage msg, Source source) {
...
         checkScriptElementsMatches(source);
         checkCssElementsMatches(msg, source);
+        checkSimpleDomMatches(msg);
         checkDomElementMatches(msg);
     }
 
     private void checkCssElementsMatches(HttpMessage msg, Source source) {
         for (AppPattern appPattern : currentApp.getCss()) {
-            if (ResourceIdentificationUtils.isCss(msg)) {
+            if (msg.getRequestHeader().isCss() || msg.getResponseHeader().isCss()) {
```
Comment (thc202) [inline addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java:131]: Revert?

File: addOns/wappalyzer/src/test/resources/org/zaproxy/zap/extension/wappalyzer/apps.json [hunk 2]
```diff
@@ -44,6 +44,16 @@
       "description": "Test Entry is a test entry for UnitTests",
       "url": "^https?://test\\.example\\.com",
       "website": "https://www.example.com/testentry"
+    },
+     "Test Entry2": {
+      "cats": [
+        36
+      ],
+    "dom": 
+      "script[src*='sites/g/files']",
```
Comment (thc202) [inline addOns/wappalyzer/src/test/resources/org/zaproxy/zap/extension/wappalyzer/apps.json:53]: Could this be reformatted? It's hard to read.

**Comment by thc202** [comment]
There's a typo in the class names (`SC`).

File: addOns/wappalyzer/src/test/resources/org/zaproxy/zap/extension/wappalyzer/apps.json (hunk 2 above)
Comment (kingthorin) [reply addOns/wappalyzer/src/test/resources/org/zaproxy/zap/extension/wappalyzer/apps.json:53]: You mean just to put it on the same line?

Comment (thc202) [reply addOns/wappalyzer/src/test/resources/org/zaproxy/zap/extension/wappalyzer/apps.json:53]: yes, and indent to the right.

**Comment by kingthorin** [comment]
Okay, I think I've fixed all those.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**